### ISSUE TITLE
JSON.stringify doesn't work for external objects

### ISSUE THREAD

**Issue by alveko** [nginx/njs#69]
When trying to call ```JSON.stringify(request)```, where ```request``` is an external object, the result is ```null```.

**Comment by xeioex** [comment]
Hi @alveko, 

You can use a helper function `njs.dump()` to see some details of the request object.

The problem with externals is that as of now, they are semi-Objects. It is not possible to used them interchangeably with native js Objects. 

In the future I plan to get rid of the external type, converting the external prototypes to native Objects. But, that is not that easy.

**Comment by reyou** [comment]
Hello @xeioex , 

Sorry to bombard you with questions today, but how do we supposed to use `njs.dump()`?

I tried following ways, but could not figure out how to read content:

```
njs.dump();
req.error(njs.dump());
req.error(JSON.stringfy(njs.dump()));
```

Could you please help?

**Comment by reyou** [comment]
This works great, thanks a lot!

**Comment by xeioex** [comment]
closed as duplicate of #16

**Comment by xeioex** [comment]
Since https://github.com/nginx/njs/commit/0fa4360a5965b66a9055d3b0cd1170faa758365e (0.3.8), `JSON.stringify()` for externals is supported.
